melview.py

```diff
@@ -119,15 +119,24 @@ def get_cookie(self):
 
 # ---------------------------------------------------------------
 
+class MelViewZone:
```

Comment:
I think having a class here is unnecessary given it's just data encapsulation - see my comment below on where you use the class.

Reply:
I do like the encapsulation. I don't want the HA parts of the code to know anything about the JSON format that mitsubishi uses

Reply:
I don't think it makes much difference - assuming there's only the below keys in the JSON object.

It's effectively using a class where a dictionary is sufficient and just adds overhead for initialisation. If you bound the on and off methods for the zone to the class itself I could understand, but as current I'm not a fan.

In regards to HA seeing the JSON data, the entire object is available in `_json` anyway, plus they already have login credentials in the config. In my opinion that's a non-issue.

Reply:
Dict is certainly sufficient, but think about how the caller needs to know something about what's in the dict (i.e.) the keys in order to read values from it.
Also think about how more maintainable the encapsulation is. If mitsubishi decide to change how zones are represented in the json, you don't want to be changing your code everywhere because of that. You would just be changing it one place.

Reply:
I can understand the encapsulation helping if there are API chaanges.
That said, the only place it would need to be changed is the refresh function where it's pulled out of `_json`.
Further, the API hasn't changed since I started this repo - besides the version number and header key.

Reply:
let's keep talking in the other comment thread -- resolving this for now